assets/semgrep_rules/client/cryptography-random-usage.cpp

```diff
@@ -0,0 +1,86 @@
+// Test cases for cryptography-random-usage rule
+#include <cstdlib>
+#include <random>
+#include "base/rand_util.h"
+#include "crypto/random.h"
+#include "third_party/boringssl/src/include/openssl/rand.h"
+#include "third_party/boringssl/src/include/openssl/evp.h"
+
+class CryptoUsageExamples {
+ public:
+  void BadRandomUsage() {
+    // SHOULD TRIGGER: Weak C-style random functions (insecure)
+    // ruleid: chromium-cryptography-random-usage
+    int weak_random1 = rand();
+    // ruleid: chromium-cryptography-random-usage
+    srand(time(nullptr));
+    // ruleid: chromium-cryptography-random-usage
+    int weak_random2 = random();
+    // ruleid: chromium-cryptography-random-usage
+    srandom(12345);
+    
+    // SHOULD TRIGGER: C++ std random functions (potentially weak)
+    // ruleid: chromium-cryptography-random-usage
+    int weak_random3 = std::rand();
+    // ruleid: chromium-cryptography-random-usage
+    std::srand(42);
+  }
+  
+  void ChromiumRandomUsage() {
+    // SHOULD TRIGGER: Chromium random functions (need security review)
+    // ruleid: chromium-cryptography-random-usage
+    int random_int = base::RandInt(1, 100);
+    // ruleid: chromium-cryptography-random-usage
+    uint64_t random_uint64 = base::RandUint64();
+    
+    uint8_t buffer[32];
+    // ruleid: chromium-cryptography-random-usage
+    base::RandGenerator(sizeof(buffer));
+  }
+  
+  void CryptographicUsage() {
+    // SHOULD TRIGGER: Cryptographic random functions (need security review)
+    uint8_t crypto_buffer[32];
+    // ruleid: chromium-cryptography-random-usage
+    crypto::RandBytes(crypto_buffer, sizeof(crypto_buffer));
+    
+    // SHOULD TRIGGER: OpenSSL random functions (need security review)
+    uint8_t ssl_buffer[16];
+    // ruleid: chromium-cryptography-random-usage
+    RAND_bytes(ssl_buffer, sizeof(ssl_buffer));
+    
+    // SHOULD TRIGGER: Encryption/Decryption operations (need security review)
+    // ruleid: chromium-cryptography-random-usage
+    EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
+    // ruleid: chromium-cryptography-random-usage
+    EVP_EncryptInit(ctx, EVP_aes_256_gcm(), key_, iv_);
+    // ruleid: chromium-cryptography-random-usage
+    EVP_DecryptInit(ctx, EVP_aes_256_gcm(), key_, iv_);
+  }
+  
+  void AcceptableUsage() {
+    // SHOULD NOT TRIGGER: Non-crypto random usage patterns
+    std::random_device rd;
+    std::mt19937 gen(rd());
```

Comment:
this should be banned. random in Chromium should be taken either from `crypto/` or from `base/`.
> Do not use any random number engines or generators from `<random>`. Instead, use `base::RandomBitGenerator`. (You may use the distributions from `<random>`.)

https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#engines-and-generators-from-random_banned